### DISCUSSION TITLE
stable-diffusion-2  text_encoder rtx 3060 12gb

### DISCUSSION THREAD

**Discussion by dsienra** [smy20011/dreambooth-gui#85] [General]
I tried to train SD V2 using text encoder but I get an error.

RuntimeError: CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 12.00 GiB total capacity; 11.23 GiB already allocated; 0 bytes free; 11.29 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

It works perfect on versions 1.4 and 1.5

I get much better results using text_encoder enable.
This is my config for versions 1.4 and 1.5

--mixed_precision=fp16
--train_batch_size=1
--gradient_accumulation_steps=1
--use_8bit_adam
--resolution=512
--gradient_checkpointing
--train_text_encoder

Learning Rate
1e-6

I tried SD V2 with all available models with the same result.

stabilityai/stable-diffusion-2-base 512
stabilityai/stable-diffusion-2-1-base 512
stabilityai/stable-diffusion-2 768
stabilityai/stable-diffusion-2-1 768

It work with the default parameters text encoder disabled.

--mixed_precision=fp16
--train_batch_size=1
--gradient_accumulation_steps=1
--use_8bit_adam

 But the model ends up broken producing just noise or terrible images,

**Comment by cheezecrisp** [comment]
how to train with  text encoder? I just edited the training png files name as instance prompt and tried "--train_text_encoder", but got an error.

**Reply by dsienra** [reply]
just using this config and train as usual

--mixed_precision=fp16
--train_batch_size=1
--gradient_accumulation_steps=1
--use_8bit_adam
--resolution=512
--gradient_checkpointing
--train_text_encoder
